src/safepoint.c

```diff
@@ -124,6 +124,14 @@ int jl_safepoint_start_gc(void)
         jl_safepoint_wait_gc();
         return 0;
     }
+    // Foreign thread adoption disables the GC and waits for it to finish, however, that may
+    // introduce a race between it and this thread checking if the GC is enabled and only
+    // then setting jl_gc_running. To avoid that, check again now that we won that race.
+    if (jl_atomic_load_acquire(&jl_gc_disable_counter)) {
+        jl_atomic_store_release(&jl_gc_running, 0);
```

Comment:
This does not seem logical to be a release, since there are no other stores in this function to act upon

Reply:
I looks like you intended the much more expensive seq-cst semantics so that it establishes an atomic ordering with the prior load?

Reply:
I'm not sure. The load during `jl_adopt_thread` is an atomic acquire, which I intended to match with this release store. I'm not sure why it would need to be matched up with stores in this very function?